packages/langium/src/grammar/references/langium-grammar-references.ts

```diff
@@ -0,0 +1,182 @@
+        if (interfaceNode) {
+            const interfaces = collectChildrenTypes(interfaceNode, this, this.documents, this.nodeLocator);
+            const targetRules: Array<ParserRule | Action> = [];
+            interfaces.forEach(interf => {
+                const rules = this.findLocalRulesWithReturnType(interf, rootNode);
+                targetRules.push(...rules);
+            });
+            if (equalURI(getDocument(target).uri, getDocument(rootNode).uri)) {
+                const ref = this.getReferenceToSelf(target);
+                if (ref) {
+                    refs.push(ref);
+                }
+            }
+            targetRules.forEach(rule => {
+                const assignment = isParserRule(rule) ?
+                    extractAssignments(rule.definition).find(a => a.feature === target.name)
+                    : getContainerOfType(rule, isGroup)!.elements.find(el => isAssignment(el) && el.feature === target.name);
+                if (assignment?.$cstNode) {
+                    const leaf = findLeafNodeAtOffset(assignment.$cstNode!, assignment.$cstNode!.offset);
+                    if (leaf) {
+                        refs.push({
+                            sourceUri: getDocument(assignment).uri,
+                            sourcePath: this.nodeLocator.getAstNodePath(assignment),
+                            targetUri: getDocument(target).uri,
+                            targetPath: this.nodeLocator.getAstNodePath(target),
+                            segment: toDocumentSegment(leaf),
+                            local: equalURI(getDocument(assignment).uri, getDocument(target).uri)
+                        });
+                    }
+                }
+            });
+        }
+        return stream(refs);
+    }
+
+    protected findReferencesToTypeAttribute(target: TypeAttribute): Stream<ReferenceDescription> {
+        const refs: ReferenceDescription[] = [];
+        const interfaceNode = getContainerOfType(target, isInterface);
+        if (interfaceNode) {
+            const ref = this.getReferenceToSelf(target);
+            if (ref) {
+                refs.push(ref);
+            }
+            const interfaces = collectChildrenTypes(interfaceNode, this, this.documents, this.nodeLocator);
+            const targetRules: Array<ParserRule | Action> = [];
+            interfaces.forEach(interf => {
+                const rules = this.findRulesWithReturnType(interf);
+                targetRules.push(...rules);
+            });
+            targetRules.forEach(rule => {
+                const assignment = isParserRule(rule) ?
+                    extractAssignments(rule.definition).find(a => a.feature === target.name)
+                    : getContainerOfType(rule, isGroup)!.elements.find(el => isAssignment(el) && el.feature === target.name);
+                if (assignment?.$cstNode) {
+                    const leaf = findLeafNodeAtOffset(assignment.$cstNode!, assignment.$cstNode!.offset);
```

Comment:
Suggestion: Same here
```suggestion
                    const leaf = findNodeForFeature(assignment.$cstNode, 'feature');
```